src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java

```diff
@@ -0,0 +1,49 @@
+package org.fogbowcloud.saps.engine.core.catalog;
+
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.Date;
+import java.util.List;
+
+import org.fogbowcloud.saps.engine.core.catalog.exceptions.CatalogException;
+import org.fogbowcloud.saps.engine.core.catalog.exceptions.TaskNotFoundException;
+import org.fogbowcloud.saps.engine.core.catalog.exceptions.UserNotFoundException;
+import org.fogbowcloud.saps.engine.core.dispatcher.notifier.Ward;
+import org.fogbowcloud.saps.engine.core.model.SapsImage;
+import org.fogbowcloud.saps.engine.core.model.SapsUser;
+import org.fogbowcloud.saps.engine.core.model.enums.ImageTaskState;
+
+public interface Catalog {
+
+    SapsImage addTask(String taskId, String dataset, String region, Date date, int priority, String user,
+                      String inputdownloadingPhaseTag, String digestInputdownloading, String preprocessingPhaseTag,
+                      String digestPreprocessing, String processingPhaseTag, String digestProcessing) throws CatalogException;
+
+    void addStateChangeTime(String taskId, ImageTaskState state, Timestamp timestamp) throws CatalogException;
+
+    void addUser(String userEmail, String userName, String userPass, boolean userState, boolean userNotify,
+                 boolean adminRole) throws CatalogException;
+
+    void addUserNotification(String submissionId, String taskId, String userEmail) throws CatalogException;
+
+    List<Ward> getUsersToNotify() throws CatalogException;
+
```

Comment:
this updateImageTask smeels bad. It seems we use it in a single line of our base, `UpdateTaskRetry`, to do the following
```
task.setUpdateTime(imageStore.getTaskById(task.getTaskId()).getUpdateTime());
imageStore.updateImageTask(task);
```

it is ugly because this catalog interface has some method to `SapsImage` object (e.g.  `addStateChangeTime`) but it is also possible to change the state of the catalog by modifying the object and calling the `updateImageTask` method

Reply:
This is a bit confusing, we should review a good design for this situation.